integration/spark/src/main/scala/org/apache/carbondata/cleanfiles/CleanFilesUtil.scala

```diff
@@ -0,0 +1,416 @@
+    var res = cleanFilesDryRunOp(carbonTable, sparkSession)
+    res ++= trashFolderDryRun(carbonTable)
+    // check for index tables
+    val indexTables = CarbonIndexUtil
+      .getIndexCarbonTables(carbonTable, sparkSession)
+    indexTables.foreach { indexTable =>
+        res ++= cleanFilesDryRunOp(indexTable, sparkSession)
+        res ++= trashFolderDryRun(indexTable)
+    }
+  res
+  }
+
+
+  /**
+   * Dry run operation for stale segments
+   */
+  def dryRunStaleSegments(carbonTable: CarbonTable, sparkSession: SparkSession): Seq[Row] = {
+    // stale segments are those segments whose entry is not present in the table status file, but
+    // those segments are in Fact folder and metadata folder
+
+    val metaDataLocation = carbonTable.getMetadataPath
+    val details = SegmentStatusManager.readLoadMetadata(metaDataLocation)
+    val finalSegments = new java.util.ArrayList[Row]()
+
+    val partitionPath = CarbonTablePath.getPartitionDir(carbonTable.getTablePath)
+    if (FileFactory.isFileExist(partitionPath)) {
+      val allSegments = FileFactory.getCarbonFile(partitionPath).listFiles
+      // there is no segment
+      if (allSegments == null || allSegments.isEmpty) Seq.empty
+      // there is no segment or failed to read tablestatus file.
+      if (details == null || details.isEmpty) Seq.empty
+      val staleSegments = TableProcessingOperations.getStaleSegments(details, allSegments, false)
+      staleSegments.asScala.foreach(
+        staleSegmentName => finalSegments.add(Row(carbonTable.getTableName, staleSegmentName._2,
+          "UNKNOWN", "FACT", "MOVE TO TRASH"))
+      )
+    } else {
+      // for partition table flow
+      val partitionFolderList : java.util.ArrayList[String] = new util.ArrayList()
+      val allFolders = FileFactory.getCarbonFile(carbonTable.getTablePath).listFiles()
+      allFolders.toSeq.foreach( name =>
+        if (name.getName.contains(CarbonCommonConstants.EQUALS) ) {
+          partitionFolderList.add(name.getAbsolutePath)
+        })
+      val tableLoadDetails = details.map(detail => detail.getLoadName)
+      partitionFolderList.asScala.foreach(name =>
+        FileFactory.getCarbonFile(name).listFiles().foreach(fileName =>
+          if (fileName.getName.endsWith(CarbonTablePath.MERGE_INDEX_FILE_EXT) && !tableLoadDetails
+            .contains(fileName.getName.substring(0, fileName.getName.indexOf(CarbonCommonConstants
+              .UNDERSCORE)))) {
+            val filePath = fileName.getAbsolutePath.split(CarbonCommonConstants.FILE_SEPARATOR)
+            finalSegments.add(Row(carbonTable.getTableName, fileName.getName.substring(0, fileName
+              .getName.indexOf("_")), "UNKNOWN", filePath(filePath.length - 2), "MOVE TO TRASH"))
+          }
+        )
+      )
+    }
+    finalSegments.asScala
+  }
+
+  /**
+   * Dry run operation for trash folder
+   */
+  def trashFolderDryRun(carbonTable: CarbonTable): Seq[Row] = {
```

Comment:
will it return all content in trash?

Reply:
yes, it will return all the segments in trash folder and show how much time is left for each for expiration.